### PR TITLE
MM-52297 Fix reactions disappearing with search open and add testing utilities

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
The fix for this issue is straightforward enough since I think some of the boolean logic was just incorrect before, but I ended up trying to add unit tests for that which made me realize that using React Testing Library with Redux, React Router, and React I18n together was painful. So I added a new `renderWithFullContext` helper which:
1. Supports all 3 of those libraries and leaves the naming open so we can add more in the future if we need to
2. Overwrites the rerender method on the object returned from React Testing Library because trying to use that on one returned by `renderWithIntl` or `renderWithIntlAndStore` rerenders the component without the extra providers added by the helper
3. Has some extra helpers that I didn't end up needing which let you modify the Redux state after the initial mount

#### Ticket Link
MM-52297

#### Screenshots
<details>
<summary>Screenshots</summary>

Centre channel with search:
Screen Shot 2023-05-25 at 5 51 30 PM

Centre channel with saved posts:
Screen Shot 2023-05-25 at 5 51 37 PM

Centre channel with pinned posts:
Screen Shot 2023-05-25 at 5 51 41 PM

Centre channel with RHS:
Screen Shot 2023-05-25 at 5 51 45 PM

</details>

#### Release Note
```release-note
Fixed post reactions disappearing when the search sidebar is open
```

### CODE REVIEW

File: webapp/channels/src/images/logo_compact.png
Comment (hmhealey) [inline webapp/channels/src/images/logo_compact.png:1]: One of those unused props was the only thing using this image, so I removed it as well

File: webapp/channels/src/tests/react_testing_utils.tsx [hunk 1]
```diff
@@ -25,3 +37,66 @@ export const renderWithIntlAndStore = (component: React.ReactNode | React.ReactN
         </IntlProvider>,
     );
 };
+
+export const renderWithFullContext = (component: React.ReactNode | React.ReactNodeArray, initialState: DeepPartial<GlobalState> = {}, locale = 'en') => {
+    // We use a redux-mock-store store for testing, but we set up a real store to ensure the initial state is complete
+    const testState = configureStore(initialState).getState();
+
+    // Store these in an object so that they can be maintained through rerenders
```
Comment (hmhealey) [inline webapp/channels/src/tests/react_testing_utils.tsx:45]: I made sure to write unit tests to confirm that this actually worked how I wanted. I'm pretty sure that, if I didn't use an object to wrap these, the arrow functions used for `rerender` and such would've captured their original values and made it impossible to update them.

In retrospect, I feel like I could've a class for this and saved these as member variables, but I'm not going to bother changing that now

**Comment by saturninoabril** [comment]
Thanks @hmhealey! Could help check if the [failed tests](https://automation-dashboard.vercel.app/cycle/340029) at `tests/integration/channels/messaging/save_post_spec.js` were regressions or not?

**Comment by hmhealey** [comment]
As far as I can tell, it's not a regression because the test is passing for me locally.

Looking at the screenshots in the Cypress report, it's confirming that there's no flagged post header visible on the saved post, and it's incorrectly detecting the pinned post header as that. I think that pinned post is left over from another test, so that'd explain why it's flaky. I can submit a fix for that, but I'll do it in a separate PR

**Review by saturninoabril — APPROVED** [review]
Thanks @hmhealey, tested and passed. LGTM.

File: webapp/channels/src/components/post/post_component.test.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                        [baseProps.post.id]: {
+                            [`${baseProps.currentUserId}-taco`]: TestHelper.getReactionMock({emoji_name: 'taco'}),
+                        },
+                    },
+                },
+            },
+        };
+
+        test('should show reactions in the center channel', () => {
+            renderWithFullContext(
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/post/post_component.test.tsx:60]: i am already liking this helper 💘

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**